### PR TITLE
Serialization enhancements in the IMap

### PR DESCRIPTION
**Pull request by tombujok**

### CODE REVIEW

File: hazelcast/src/main/java/com/hazelcast/map/impl/operation/ClearBackupOperation.java [hunk 1]
```diff
@@ -16,11 +16,13 @@
 
 package com.hazelcast.map.impl.operation;
 
-import com.hazelcast.nio.serialization.DataSerializable;
+import com.hazelcast.map.impl.MapDataSerializerHook;
+import com.hazelcast.nio.serialization.IdentifiedDataSerializable;
 import com.hazelcast.spi.BackupOperation;
 import com.hazelcast.spi.impl.MutatingOperation;
 
-public class ClearBackupOperation extends MapOperation implements BackupOperation, MutatingOperation, DataSerializable {
```
Comment (pveentjer) [inline hazelcast/src/main/java/com/hazelcast/map/impl/operation/ClearBackupOperation.java]: Isn't it easier to impose it on MapOperation? This way you know you are going to catch all map operations.

**Comment by pveentjer** [comment]
It looks good. Please have a look at the comment regarding moving the logic into the MapOperation.

Can you also squash the commits?

**Comment by pveentjer** [comment]
If IdentifiedDataSerializable is not moved into MapOperation that is ok with me. However for the rolling upgrade feature it would be smart to convert every DataSerializable implementation we have (not only operations) to IdentifiedDataSerializable since the latter reduces coupling with the implementation since the class name is not part of the byte-stream.

**Comment by tombujok** [comment]
I'll do it on the MapOperation level

**Comment by tombujok** [comment]
@pveentjer all operations are IdentifiedDataSerializable now. Will be squashed at merge. Could you have a look?

**Comment by pveentjer** [comment]
:+1:

**Comment by pveentjer** [comment]
Since we have 2 :+1: , I'm merging this one

**Pull request merged**